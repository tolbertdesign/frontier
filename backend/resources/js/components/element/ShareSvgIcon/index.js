import ShareSvgIcon from './ShareSvgIcon.vue'

export default ShareSvgIcon
