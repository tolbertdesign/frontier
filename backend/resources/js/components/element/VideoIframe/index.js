import VideoIframe from './VideoIframe.vue'

export default VideoIframe
