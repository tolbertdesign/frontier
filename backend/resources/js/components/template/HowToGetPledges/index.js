import HowToGetPledges from './HowToGetPledges.vue'

export default HowToGetPledges
