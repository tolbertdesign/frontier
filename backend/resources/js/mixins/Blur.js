export default {
  methods: {
    blur () {
      document.getElementById('app').style.filter = 'blur(4px)'
    },
    unBlur () {
      document.getElementById('app').style.filter = 'none'
    },
  },
}
