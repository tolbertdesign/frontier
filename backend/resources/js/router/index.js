import Vue from 'vue'
import VueRouter from 'vue-router'
import AppContainer from '@/layouts/AppContainer'
import NProgress from 'nprogress'
import ParentDashboard from '@/views/ParentDashboard'
import TeacherDashboard from '@/views/TeacherDashboard'
import EditParticipant from '@/views/EditParticipant'
import EditProfile from '@/views/EditProfile'
import EasyEmailer from '@/views/EasyEmailer/EasyEmailer'
import FinishLine from '@/views/FinishLine'
import store from '@/store'

Vue.use(VueRouter)

const routes = [
  {
    path: '/v3/home',
    component: AppContainer,
    children: [
      {
        path: 'dashboard',
        name: 'parent-dashboard',
        component: ParentDashboard,
      },
      {
        path: 'teacher-dashboard',
        name: 'teacher-dashboard',
        component: TeacherDashboard,
      },
      {
        path: 'edit-participant/:id',
        name: 'edit-participant',
        component: EditParticipant,
      },
      {
        path: 'edit-profile',
        name: 'edit-profile',
        component: EditProfile,
      },
      {
        path: 'easy-emailer/:participantUserId',
        name: 'easy-emailer',
        component: EasyEmailer,
      },
      {
        path: 'finish-line/:participantUserId',
        name: 'finish-line',
        component: FinishLine,
      },
    ],
  },

]

const router = new VueRouter({
  mode: 'history',
  forcePageRouteRefresh: false,
  linkActiveClass: 'is-active',
  routes,

  scrollBehavior (to, from, savedPosition) {
    return { x: 0, y: 0 }
  },
})

router.beforeEach((routeTo, routeFrom, next) => {
  NProgress.start()
  store.dispatch('setContentGroup', routeTo)

  if (router.options.forcePageRouteRefresh) {
    window.location.href = '/v3/home/dashboard/' + routeTo.path
    router.options.forcePageRouteRefresh = false
  } else {
    next()
  }
})

router.afterEach(() => {
  NProgress.done()
})

export default router
