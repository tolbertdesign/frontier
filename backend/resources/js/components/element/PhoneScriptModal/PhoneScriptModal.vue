<template>
  <div class="phone-script-modal m-4 rounded-xl overflow-hidden shadow-lg">
    <div class="bg-white">
      <header class="modal-card-head bg-white relative items-start">
        <p class="font-bold modal-card-title text-center text-xl">
          {{ lang.header }}
          <!-- <span class="script-header text-center">{{ lang.header }}</span> -->
        </p>
      </header>
      <section class="modal-card-body has-text-centered">
        <p
          class="script-body"
          v-html="lang.p1"
        />
        <br>
        <p
          class="script-body"
          v-html="lang.p2"
        />
        <br>
        <p
          class="script-body"
          v-html="parseLanguage(lang.p3, {event_name: program.event_name})"
        />
        <br>
        <p
          class="script-body"
          v-html="parseLanguage(lang.p4, {event_name: program.event_name,
                                          range_low: program.unit_range_low,
                                          range_high: program.unit_range_high,
                                          unit_max: program.unit_max_charge,
                                          unit_types: program.unit.name_plural})"
        />
        <br>
        <p
          class="script-body"
          v-html="lang.p5"
        />
        <br>
        <p
          class="script-body"
          v-html="parseLanguage(lang.p6, {unit_type: program.unit.name.toUpperCase()})"
        />
        <br>
        <p
          class="script-body"
          v-html="parseLanguage(lang.p7, {unit_type: program.unit.name})"
        />
        <br>
        <p
          class="script-body"
          v-html="parseLanguage(lang.p8, {unit_types: program.unit.name_plural})"
        />
        <br>
        <p
          class="script-body"
          v-html="lang.p9"
        />
        <br>
        <button
          class="px-8 rounded-full button inline-block is-secondary shadow-md font-bold centered close-button"
          @click="close()"
        >
          {{ lang.close }}
        </button>
      </section>
    </div>
  </div>
</template>

<script>
import Blur from '@/mixins/Blur'

export default {
  name: 'PhoneScriptModal',
  mixins: [Blur],
  props: {
    program: {
      type: Object,
      default: () => {},
    },
  },
  computed: {
    lang () {
      return this.$store.state.lang.how_to_get_pledges.phone_script
    },
  },
  methods: {
    close () {
      this.$emit('close')
      this.unBlur()
    },
  },
}
</script>
