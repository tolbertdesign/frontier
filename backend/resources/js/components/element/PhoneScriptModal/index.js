import PhoneScriptModal from './PhoneScriptModal.vue'

export default PhoneScriptModal
