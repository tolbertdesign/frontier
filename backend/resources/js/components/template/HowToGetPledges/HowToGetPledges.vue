<template>
  <div class="card-content">
    <div>
      <div class="pb-4">
        <figure class="image is-16by9 mb-4 border">
          <VideoIframe
            v-if="video"
            :video-id="video.external_video_id"
            :source="video.source"
          />
        </figure>
      </div>

      <h4 class="font-bold text-lg text-center md:text-left mb-4">
        {{ lang.asking_for_pledges }}
      </h4>
      <ul class="list-reset text-sm md:text-xl">
        <li class="flex items-center mb-2">
          <RouterLink
            :to="{
              name: 'easy-emailer',
              params: {
                participantUserId: firstParticipantId
              }
            }"
            strict
            class="font-semibold underline"
          >
            <ShareSvgIcon type="email" />
          </RouterLink>
          <p class="ml-4">
            <RouterLink
              :to="{
                name: 'easy-emailer',
                params: {
                  participantUserId: firstParticipantId
                }
              }"
              strict
              class="font-semibold underline"
            >
              {{ lang.email.link }}
            </RouterLink> {{ lang.email.paragraph }}
          </p>
        </li>
        <li class="flex items-center mb-2">
          <button @click="shareFacebook()">
            <ShareSvgIcon type="facebook" />
          </button>
          <p class="ml-4">
            <a
              class="font-semibold underline"
              @click.prevent="shareFacebook()"
            >
              {{ lang.share_on_facebook }}
            </a> {{ lang.get_sponsor_pledges }}
          </p>
        </li>
        <li
          v-if="!program.ssv_disabled"
          class="flex items-center mb-2"
        >
          <a href="#">
            <ShareSvgIcon type="ssv" />
          </a>
          <p class="ml-4">
            <a
              href="#"
              class="font-semibold underline"
              @click.prevent="scrollTo()"
            >
              {{ lang.share_ssv }}
            </a> {{ lang.customize_it }}
          </p>
        </li>
        <li class="flex items-center mb-2">
          <a
            href="#"
            class="phone-script-icon"
            @click.prevent="openPhoneScript"
          >
            <ShareSvgIcon type="script" />
          </a>
          <p class="ml-4">
            <a
              href="#"
              class="font-semibold underline phone-script-link"
              @click.prevent="openPhoneScript"
            >
              {{ lang.use_our_phone_script }}
            </a> {{ lang.call_friends }}
          </p>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
import PhoneScriptModal from '@/components/element/PhoneScriptModal'
import ShareSvgIcon from '@/components/element/ShareSvgIcon'
import VideoIframe from '@/components/element/VideoIframe'
import Blur from '@/mixins/Blur'
import axios from 'axios'

export default {
  name: 'HowToGetPledges',
  status: 'prototype',
  release: '1.0.0',
  components: {
    ShareSvgIcon,
    VideoIframe,
  },
  mixins: [Blur],
  props: {
    program: {
      type: Object,
      default: null,
    },
  },
  data () {
    return {
      video: null,
    }
  },
  computed: {
    lang () {
      return this.$store.state.lang.how_to_get_pledges
    },
    firstParticipantId () {
      return this.program.participants[0].id
    },
    hasVideo () {
      return this.program.participants.reduce((result, participant) => {
        const hasImage = !!participant.profile.image_name
        return result || hasImage
      }, false)
    },
    specialUrls () {
      return this.program.participants[0].special_urls
    },
  },
  created () {
    this.fetchVideos()
  },
  methods: {
    isDesktop () {
      return window.matchMedia('(min-width: 769px)').matches
    },
    scrollTo () {
      this.$parent.$parent.$parent.items[2].isOpen = true

      setTimeout(() => {
        // scroll to top
        $('html, body').animate({
          scrollTop: $(this.$parent.$parent.$parent.$el).offset().top,
        }, 300)
      }, 310)
    },

    fetchVideos (videoMetadata) {
      axios.get('/v3/api/videos/get-pledges/' + this.program.id).then(response => {
        this.video = response.data
      }).catch(error => {
        console.error(error)
      })
    },
    openPhoneScript (evt) {
      this.$buefy.modal.open({
        parent: this,
        component: PhoneScriptModal,
        hasModalCard: false,
        canCancel: ['close', 'outside'],
        onCancel: this.unBlur,
        props: { program: this.program },
      })

      this.blur()

      this.gaEvent('Parent Dashboard', 'How To Get Pledges', 'Phone Script')
    },
    getShareUrl (referrerName) {
      const filterFunction = function (specialUrl) {
        return specialUrl.referrer.name === referrerName
      }
      const shortKey = this.specialUrls.filter(filterFunction)[0].short_key
      return window.location.protocol + '//' + window.location.hostname + '/v3/dash/' + shortKey
    },
    shareFacebook () {
      const referrerName = this.hasVideo ? 'Facebook_Video' : 'Facebook'
      const url = this.getShareUrl(referrerName)
      const participantUserId = this.program.participants[0].id
      FB.ui(
        {
          method: 'share',
          href: url,
        },
        function (response) {
          if (response && !response.error_message) {
            axios.get('/users/user_successful_facebook_share/' + participantUserId)
              .catch(console.log)
            var label = this.hasVideo ? 'Pledge Success Page Video' : 'Pledge Success Page'
            ga('send', 'event', 'Share Public Pledge Page', 'Click Facebook Share', label, 1)
          }
        },
      )
    },
  },
}
</script>
