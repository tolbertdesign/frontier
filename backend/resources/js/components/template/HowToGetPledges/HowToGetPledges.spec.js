import { shallowMount, createLocalVue } from '@vue/test-utils'
import HowToGetPledges from './HowToGetPledges.vue'
import PhoneScriptModal from '@/components/element/PhoneScriptModal'
import ShareSvgIcon from '@/components/element/ShareSvgIcon'
import Blur from '@/mixins/Blur'
import router from '@/router'
import VideoIframe from '@/components/element/VideoIframe'

const localVue = createLocalVue()

const $store = {
  state: {
    lang: {
      how_to_get_pledges: {
        asking_for_pledges: 'asking for pledges',
        email: {
          link: 'link',
          paragraph: 'paragraph',
        },
        share_on_facebook: 'share on facebook',
        get_sponsor_pledges: 'get sponsor pledges',
        share_ssv: 'share ssv',
        customize_it: 'customize it',
        use_our_phone_script: 'use our phone script',
        call_friends: 'call friends',
      },
    },
  },
}

const program = {
  id: 99,
  ssv_disabled: 0,
  participants: [
    {
      id: 1,
      special_urls: 'special URLs 1',
    },
    {
      id: 2,
      special_urls: 'special URLs 2',
    },
    {
      id: 3,
      special_urls: 'special URLs 3',
    },
  ],
}

describe('HowToGetPledges', () => {
  it('Snapshop test for HowToGetPledges', () => {
    const wrapper = shallowMount(HowToGetPledges, {
      localVue,
      router,
      mixins: {
        Blur,
      },
      components: {
        PhoneScriptModal,
        ShareSvgIcon,
        VideoIframe,
      },
      mocks: {
        $store,
      },
      propsData: {
        program: program,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('Snapshop test for HowToGetPledges with ssv disabled', () => {
    program.ssv_disabled = 1
    const wrapper = shallowMount(HowToGetPledges, {
      localVue,
      router,
      mixins: {
        Blur,
      },
      components: {
        PhoneScriptModal,
        ShareSvgIcon,
        VideoIframe,
      },
      mocks: {
        $store,
      },
      propsData: {
        program: program,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
