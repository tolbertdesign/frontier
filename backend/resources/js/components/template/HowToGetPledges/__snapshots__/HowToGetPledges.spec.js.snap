// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToGetPledges Snapshop test for HowToGetPledges 1`] = `
<div class="card-content">
  <div>
    <div class="pb-4">
      <figure class="image is-16by9 mb-4 border">
        <!---->
      </figure>
    </div>
    <h4 class="font-bold text-lg text-center md:text-left mb-4">
      asking for pledges
    </h4>
    <ul class="list-reset text-sm md:text-xl">
      <li class="flex items-center mb-2">
        <routerlink-stub to="[object Object]" tag="a" event="click" strict="" class="font-semibold underline">
          <sharesvgicon-stub type="email"></sharesvgicon-stub>
        </routerlink-stub>
        <p class="ml-4">
          <routerlink-stub to="[object Object]" tag="a" event="click" strict="" class="font-semibold underline">
            link
          </routerlink-stub> paragraph
        </p>
      </li>
      <li class="flex items-center mb-2"><button>
          <sharesvgicon-stub type="facebook"></sharesvgicon-stub>
        </button>
        <p class="ml-4"><a class="font-semibold underline">
            share on facebook
          </a> get sponsor pledges
        </p>
      </li>
      <li class="flex items-center mb-2"><a href="#">
          <sharesvgicon-stub type="ssv"></sharesvgicon-stub>
        </a>
        <p class="ml-4"><a href="#" class="font-semibold underline">
            share ssv
          </a> customize it
        </p>
      </li>
      <li class="flex items-center mb-2"><a href="#" class="phone-script-icon">
          <sharesvgicon-stub type="script"></sharesvgicon-stub>
        </a>
        <p class="ml-4"><a href="#" class="font-semibold underline phone-script-link">
            use our phone script
          </a> call friends
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`HowToGetPledges Snapshop test for HowToGetPledges with ssv disabled 1`] = `
<div class="card-content">
  <div>
    <div class="pb-4">
      <figure class="image is-16by9 mb-4 border">
        <!---->
      </figure>
    </div>
    <h4 class="font-bold text-lg text-center md:text-left mb-4">
      asking for pledges
    </h4>
    <ul class="list-reset text-sm md:text-xl">
      <li class="flex items-center mb-2">
        <routerlink-stub to="[object Object]" tag="a" event="click" strict="" class="font-semibold underline">
          <sharesvgicon-stub type="email"></sharesvgicon-stub>
        </routerlink-stub>
        <p class="ml-4">
          <routerlink-stub to="[object Object]" tag="a" event="click" strict="" class="font-semibold underline">
            link
          </routerlink-stub> paragraph
        </p>
      </li>
      <li class="flex items-center mb-2"><button>
          <sharesvgicon-stub type="facebook"></sharesvgicon-stub>
        </button>
        <p class="ml-4"><a class="font-semibold underline">
            share on facebook
          </a> get sponsor pledges
        </p>
      </li>
      <!---->
      <li class="flex items-center mb-2"><a href="#" class="phone-script-icon">
          <sharesvgicon-stub type="script"></sharesvgicon-stub>
        </a>
        <p class="ml-4"><a href="#" class="font-semibold underline phone-script-link">
            use our phone script
          </a> call friends
        </p>
      </li>
    </ul>
  </div>
</div>
`;
